tests/responses/ok-test.js

```diff
@@ -213,6 +222,168 @@ test('Does not return included data if "included = false"', function (t) {
   });
 });
 
+test('Support simple sorting', function (t) {
+  t.plan(4);
+
+  sails.request({
+    url   : '/user',
+    data  : {sort: 'firstName'},
+    method: 'GET'
+  }, function (err, res, body) {
+    if (err) {
+      t.fail(err);
+    }
+    try {
+      t.equal(res.statusCode, 200, 'HTTP status code is 200');
+      t.equal(res.headers['Content-Type'], 'application/vnd.api+json', 'Sends jsonapi mime type');
+      t.ok(validateJsonApi(body), 'Body is a valid JSON API');
+      t.deepEqual(
+        body.data.map(function (d) {
+          return d.id;
+        }),
+        ['2', '3', '1'],
+        'Data is in correct order'
+      );
+    } catch (err) {
+      t.fail(err);
+    }
+    t.end();
+  });
+});
+
+test('Support descending sorting', function (t) {
+  t.plan(4);
+
+  sails.request({
+    url   : '/user',
+    data  : {sort: '-firstName'},
+    method: 'GET'
+  }, function (err, res, body) {
+    if (err) {
+      t.fail(err);
+    }
+    try {
+      t.equal(res.statusCode, 200, 'HTTP status code is 200');
+      t.equal(res.headers['Content-Type'], 'application/vnd.api+json', 'Sends jsonapi mime type');
+      t.ok(validateJsonApi(body), 'Body is a valid JSON API');
+      t.deepEqual(
+        body.data.map(function (d) {
+          return d.id;
+        }),
+        ['1', '3', '2'],
+        'Data is in correct order'
+      );
+    } catch (err) {
+      t.fail(err);
+    }
+    t.end();
+  });
+});
+
+test('Support sorting by multiple fields', function (t) {
+  t.plan(4);
+
+  sails.request({
+    url   : '/user',
+    data  : {sort: 'lastName,firstName'},
+    method: 'GET'
+  }, function (err, res, body) {
+    if (err) {
+      t.fail(err);
+    }
+    try {
+      t.equal(res.statusCode, 200, 'HTTP status code is 200');
+      t.equal(res.headers['Content-Type'], 'application/vnd.api+json', 'Sends jsonapi mime type');
+      t.ok(validateJsonApi(body), 'Body is a valid JSON API');
+      t.deepEqual(
+        body.data.map(function (d) {
+          return d.id;
+        }),
+        ['2', '3', '1'],
+        'Data is in correct order'
+      );
+    } catch (err) {
+      t.fail(err);
+    }
+    t.end();
+  });
+});
+
+/*
+ * not supported yet
+ *
+test('Supports fetching relationships', function (t) {
+  t.plan(2);
+
+  sails.request({
+    url   : '/author/1/relationships/books',
+    method: 'GET'
+  }, function (err, res, body) {
+    if (err) {
+      t.fail(err);
+    }
+    try {
+      t.equal(res.statusCode, 200, 'HTTP status code is 200');
+      t.equal(res.headers['Content-Type'], 'application/vnd.api+json', 'Sends jsonapi mime type');
+      t.ok(validateJsonApi(body), 'Body is a valid JSON API');
+    } catch (err) {
+      t.fail(err);
+    }
+    t.end();
+  });
+});
+ */
```

Comment:
That would be great, or just make a new commit. I am not too overly picky
about the commit history. :)

On Thu, Apr 21, 2016, 08:42 jelhan <notifications@github.com> wrote:

> In tests/responses/ok-test.js
> <https://github.com/IanVS/sails-hook-jsonapi/pull/7#discussion_r60572563>:
>
> > +    method: 'GET'
> > +  }, function (err, res, body) {
> > +    if (err) {
> > +      t.fail(err);
> > +    }
> > +    try {
> > +      t.equal(res.statusCode, 200, 'HTTP status code is 200');
> > +      t.equal(res.headers['Content-Type'], 'application/vnd.api+json', 'Sends jsonapi mime type');
> > +      t.ok(validateJsonApi(body), 'Body is a valid JSON API');
> > +    } catch (err) {
> > +      t.fail(err);
> > +    }
> > +    t.end();
> > +  });
> > +});
> > + */
>
> Nice. Didn't know that one. Do you like me to edit the commit and change
> to test.skip?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/IanVS/sails-hook-jsonapi/pull/7/files/f0ad8f45cb2e8a74078327a84e13024bf7ce4eb0#r60572563>
>